internal/kafka/internal/clusters/provider_test.go

```diff
@@ -0,0 +1,118 @@
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/db"
+	. "github.com/onsi/gomega"
+)
+
+func TestNewDefaultProviderFactory(t *testing.T) {
+	type args struct {
+		ocmClient              ocm.ClusterManagementClient
+		connectionFactory      *db.ConnectionFactory
+		ocmConfig              *ocm.OCMConfig
+		awsConfig              *config.AWSConfig
+		dataplaneClusterConfig *config.DataplaneClusterConfig
+	}
+	tests := []struct {
+		name string
+		args args
+		want *DefaultProviderFactory
+	}{
+		{
+			name: "Should return the default Provider Factory",
+			args: args{},
+			want: &DefaultProviderFactory{
+				providerContainer: map[api.ClusterProviderType]Provider{
+					api.ClusterProviderStandalone: &StandaloneProvider{},
+					api.ClusterProviderOCM:        &OCMProvider{
+						clusterBuilder: &clusterBuilder{
+							idGenerator: ocm.NewIDGenerator("mk-"),
+						},
+					},
+				},
+			},
+		},
+	}
+	RegisterTestingT(t)
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := NewDefaultProviderFactory(tt.args.ocmClient, tt.args.connectionFactory, tt.args.ocmConfig, tt.args.awsConfig, tt.args.dataplaneClusterConfig)
+			Expect(got).To(Equal(tt.want))
+		})
+	}
+}
+
+func TestDefaultProviderFactory_GetProvider(t *testing.T) {
+	type fields struct {
+		providerContainer map[api.ClusterProviderType]Provider
+	}
+	type args struct {
+		providerType api.ClusterProviderType
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		want    Provider
+		wantErr bool
+	}{
+		{
+			name: "Should return ocm Provider when provider type is empty",
+			fields: fields{
+				providerContainer: map[api.ClusterProviderType]Provider{
+					api.ClusterProviderStandalone: &StandaloneProvider{},
+					api.ClusterProviderOCM:        &OCMProvider{},
+				},
+			},
+			args: args{
+				providerType: "",
+			},
+			want: &OCMProvider{},
+		},
+		{
+			name: "Should return correct Provider type",
+			fields: fields{
+				providerContainer: map[api.ClusterProviderType]Provider{
+					api.ClusterProviderStandalone: &StandaloneProvider{},
+					api.ClusterProviderOCM:        &OCMProvider{},
+				},
+			},
+			args: args{
+				providerType: "ocm",
+			},
+			want: &OCMProvider{},
+		},
+		{
+			name: "Should return error when invalid provider type is given",
+			fields: fields{
+				providerContainer: map[api.ClusterProviderType]Provider{
+					api.ClusterProviderStandalone: &StandaloneProvider{},
+					api.ClusterProviderOCM:        &OCMProvider{},
+				},
+			},
+			args: args{
+				providerType: "invalid-provider-type",
+			},
+			want: nil,
+			wantErr: true,
+		},
+	}
+	RegisterTestingT(t)
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			d := &DefaultProviderFactory{
+				providerContainer: tt.fields.providerContainer,
+			}
+			got, err := d.GetProvider(tt.args.providerType)
+			Expect(err!=nil).To(Equal(tt.wantErr))
+			if got != nil{
+			Expect(got).To(Equal(tt.want))
+			}
```

Comment:
What if `got` is `nil` but `tt.want != nil` ?
Wouldn't be better to just get rid of the `if` and leave just the `Expect` ?

Reply:
I ran into an issue where the test wasnt passing when I left the Expect there. The error it was giving me was Refusing to compare <nil> to <nil>. Wasn't sure how to get around it other than this, Have you any suggestions ?

Reply:
`Expect(got).To(Or(BeNil(),Equal(tt.want)))` this seems to solve the issue but im still happy for other suggestions that may be better.

Reply:
I believe the issue is this one: https://github.com/onsi/gomega/issues/267
Probably then the best we can do is something like this:
```go
Expect(got==nil).To(Equal(tt.want==nil))
if got != nil {
  Expect(got).To(Equal(tt.want))
}

```